ledger/sandbox/src/main/scala/com/digitalasset/platform/index/RecoveringIndexer.scala

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.index
+
+import java.util.concurrent.atomic.{AtomicBoolean, AtomicReference}
+
+import akka.actor.ActorSystem
+import akka.pattern.after
+import com.digitalasset.platform.common.util.{DirectExecutionContext => DEC}
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+object RecoveringIndexer {
+  def apply(restartDelay: FiniteDuration, asyncTolerance: FiniteDuration): RecoveringIndexer =
+    new RecoveringIndexer(restartDelay, asyncTolerance)
+}
+
+/**
+  * A helper that restarts an indexer whenever an error occurs.
+  *
+  * @param restartDelay Time to wait before restarting the indexer after a failure
+  * @param asyncTolerance Time to wait for asynchronous operations to complete
+  */
+class RecoveringIndexer(restartDelay: FiniteDuration, asyncTolerance: FiniteDuration)
+    extends AutoCloseable {
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  private[this] val actorSystem = ActorSystem("RecoveringIndexer")
```

Comment:
We don't need a separate actorsystem for this, but can rather pass in a scheduler as parameter.